xdsl/transforms/canonicalization_patterns/riscv.py

```diff
@@ -70,6 +70,20 @@ def match_and_rewrite(self, op: riscv.MulOp, rewriter: PatternRewriter) -> None:
                 return
 
 
+class MultiplyImmediateOne(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: riscv.MulOp, rewriter: PatternRewriter) -> None:
+        if (rs2 := get_constant_value(op.rs2)) is not None and rs2.value.data == 1:
+            rd_type = cast(riscv.IntRegisterType, op.rd.type)
+
+            move_op = riscv.MVOp(
+                op.rs1,
+                rd=rd_type,
+            )
+
+            rewriter.replace_op(op, [move_op])
```

Comment:
Hmm, are you sure that this isn't already satisfied by `MultiplyImmediates`? What happens when you comment out this pattern?